spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractJdbcJobSearchableExecutionDaoTests.java

```diff
@@ -0,0 +1,210 @@
+import org.springframework.batch.item.database.support.DataFieldMaxValueIncrementerFactory;
+import org.springframework.cloud.dataflow.core.database.support.DatabaseType;
+import org.springframework.cloud.dataflow.core.database.support.MultiSchemaIncrementerFactory;
+import org.springframework.cloud.task.batch.listener.support.JdbcTaskBatchDao;
+import org.springframework.cloud.task.repository.TaskRepository;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public abstract class AbstractJdbcJobSearchableExecutionDaoTests extends AbstractDaoTests {
+
+	private static final String BASE_JOB_INSTANCE_NAME = "JOB_INST_";
+
+	protected JdbcSearchableJobExecutionDao jdbcSearchableJobExecutionDao;
+
+	protected JdbcSearchableJobInstanceDao jdbcSearchableJobInstanceDao;
+
+	protected DataSource dataSource;
+
+	protected DataFieldMaxValueIncrementerFactory incrementerFactory;
+
+	protected JdbcJobExecutionDao jobExecutionDao;
+
+	protected JdbcStepExecutionDao stepExecutionDao;
+
+	protected JdbcTemplate jdbcTemplate;
+
+	protected TaskRepository taskRepository;
+
+	protected JdbcTaskBatchDao jdbcTaskBatchDao;
+
+	public void setupSearchableExecutionDaoTest(JdbcDatabaseContainer dbContainer, String schemaName,
+												DatabaseType databaseType) throws Exception {
+		this.dataSource = createDataSourceForContainer(dbContainer);
+		this.jdbcTemplate = new JdbcTemplate(this.dataSource);
+		createDataFlowSchema(dbContainer, schemaName);
+
+		this.jdbcSearchableJobExecutionDao = new JdbcSearchableJobExecutionDao();
+		this.jdbcSearchableJobExecutionDao.setDataSource(this.dataSource);
+		this.jdbcSearchableJobExecutionDao.afterPropertiesSet();
+		this.jdbcSearchableJobInstanceDao = new JdbcSearchableJobInstanceDao();
+		this.jdbcSearchableJobInstanceDao.setJdbcTemplate(this.jdbcTemplate);
+		incrementerFactory = new MultiSchemaIncrementerFactory(dataSource);
+
+		this.jdbcSearchableJobInstanceDao.setJobIncrementer(incrementerFactory.getIncrementer(databaseType.name(),
+			AbstractJdbcBatchMetadataDao.DEFAULT_TABLE_PREFIX + "JOB_SEQ"));
+		jobExecutionDao = new JdbcJobExecutionDao();
+		jobExecutionDao.setJobExecutionIncrementer(incrementerFactory.getIncrementer(databaseType.name(),
+			AbstractJdbcBatchMetadataDao.DEFAULT_TABLE_PREFIX + "JOB_EXECUTION_SEQ"));
+		this.jobExecutionDao.setJdbcTemplate(new JdbcTemplate(this.dataSource));
+		jobExecutionDao.afterPropertiesSet();
+		this.stepExecutionDao = new JdbcStepExecutionDao();
+		this.stepExecutionDao.setJdbcTemplate(this.jdbcTemplate);
+		this.stepExecutionDao.setStepExecutionIncrementer(incrementerFactory.getIncrementer(databaseType.name(),
+			AbstractJdbcBatchMetadataDao.DEFAULT_TABLE_PREFIX + "STEP_EXECUTION_SEQ"));
+		this.stepExecutionDao.afterPropertiesSet();
+	}
+
+	@Test
```

Comment:
Since these are newly added tests and using Junit5 let's take advantage of "package-protected" visibility modifiers on the test methods ("less is more" principle). Also, the names no longer require the annoying "test" prefix and instead can express more clearly what the test is doing. One naming format that seems to work well is `shouldDo<thing>When<condition>` or even just dropping the `test` prefix reads better IMO. Wdyt?